File: PHP/src/Behavioral/Strategy/ExerciseOne/NotificationStrategys/SMSNotificationStrategy.php
<?php

namespace DesignPatterns\Behavioral\Strategy\ExerciseOne\NotificationStrategys;

use DesignPatterns\Behavioral\Strategy\ExerciseOne\Interfaces\NotificationInterface;
use DesignPatterns\Behavioral\Strategy\ExerciseOne\Models\User;

class SmsNotificationStrategy implements NotificationInterface
{
    /**
     * @param User $user
     * @param String $message
     * @return void
     */
    public function sendNotification(User $user, String $message): void
    {
        echo "Sending sms to {$user->getName()} with message: {$message}";
    }
}

File: dart_lang/lib/b_behavioral/b_strategy/c_exercise/notification_strategy/strategies/sms_notification_strategy.dart
import 'package:dart_lang/b_behavioral/b_strategy/c_exercise/notification_strategy/notification_strategy_inteface.dart';
import 'package:dart_lang/b_behavioral/b_strategy/c_exercise/user.dart';

class SmsNotificationStrategy implements NotificationStrategy {
  @override
  void sendNotification(User user, String message) {
    print("sending SMS... ");
  }
}
